libraries/aws_kms_key.rb

```diff
@@ -71,4 +74,19 @@ def has_rotation_enabled?
   def to_s
     opts.key?(:aws_region) ? "KMS Key #{@display_name} in #{opts[:aws_region]}" : "KMS Key #{@display_name}"
   end
+
+  def fetch_key_id
+    catch_aws_errors do
+      key_id = ''
+      response = @aws.kms_client.list_aliases
+      return [] if !response || response.empty?
+      response.aliases.each do |alias_entry|
+        if alias_entry['alias_name'] == @alias
+          key_id = alias_entry['target_key_id'] if alias_entry['target_key_id']
+        end
+      end
+      raise 'the provided alias, does not have an associated kms key-id' if key_id.empty? || key_id.nil?
```

Comment:
Adding the return above will allow you to remove the `if` here and simply raise if execution reaches this point.

Also instead of raising here, you should throw to the base class with `fail_resource()`:
```
fail_resource("Unable to find KMS Key using provided alias: #{@alias}")
```
It will provide better formatting and context to the user upon failure.